pkg/build/api/validation/validation.go

```diff
@@ -355,6 +355,21 @@ func validateDockerStrategy(strategy *buildapi.DockerBuildStrategy) fielderrors.
 	}
 
 	allErrs = append(allErrs, validateSecretRef(strategy.PullSecret).Prefix("pullSecret")...)
+
+	if len(strategy.DockerfilePath) != 0 {
+		cleaned := path.Clean(strategy.DockerfilePath)
+		if strings.HasPrefix(cleaned, "/") {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("dockerfilePath", strategy.DockerfilePath, "dockerfilePath must not be an absolute path"))
+		} else if strings.HasPrefix(cleaned, "..") {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("dockerfilePath", strategy.DockerfilePath, "dockerfilePath must not start with .."))
+		} else {
+			if cleaned == "." {
+				cleaned = ""
+			}
+			strategy.DockerfilePath = cleaned
+		}
```

Comment:
someone's probably going to tell you this should be a switch, but it won't be me :)